### PR TITLE
core(fr): add fetcher to transitional driver

### PR DESCRIPTION
**Pull request by adamraine**
To make the robots.txt gatherer compatible with CSPs we should use the new fetcher. This PR makes fetcher available to FR gatherers.

Related
https://github.com/GoogleChrome/lighthouse/issues/10225

### CODE REVIEW

File: lighthouse-core/gather/driver.js
Comment (adamraine) [inline lighthouse-core/gather/driver.js:82]: Need `executionContext` and `defaultSession` to be defined for the Fetcher constructor.

**Review by patrickhulce — APPROVED** [review]
nice!

File: lighthouse-core/gather/driver.js
Comment (patrickhulce) [reply lighthouse-core/gather/driver.js:82]: add a comment to that effect in the code?

File: lighthouse-core/gather/fetcher.js
Comment (patrickhulce) [inline lighthouse-core/gather/fetcher.js:24]: FWIW could avoid the sticky issue of the other properties being ready/demanding a particular initialization order by saving `this._driver = driver` instead, not a big deal though

File: lighthouse-core/fraggle-rock/gather/driver.js
Comment (patrickhulce) [inline lighthouse-core/fraggle-rock/gather/driver.js:70]: I've avoided passing `this` so far, WDYT about passing `session` and `executionContext` instead?

**Review by patrickhulce — CHANGES_REQUESTED** [review]
er, actually I think I feel strongly enough about https://github.com/GoogleChrome/lighthouse/pull/12419/files#r622408557 to have a discussion about it at least before approval :)

File: lighthouse-core/gather/driver.js
Comment (adamraine) [reply lighthouse-core/gather/driver.js]: Now that `defaultSession` and `executionContext` are passed in individually, I feel it's pretty obvious on its own.

**Comment by adamraine** [comment]
I didn't feel very strongly, thought https://github.com/GoogleChrome/lighthouse/pull/12419#discussion_r622408557 was a good idea.

File: lighthouse-core/gather/driver.js
Comment (patrickhulce) [reply lighthouse-core/gather/driver.js]: Yes 100% agreed. It's not necessary with the new approach 🎉  :)

**Review by patrickhulce — APPROVED** [review]

**Pull request merged**
